kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/clusternetworkaddressconfigprovider/IndexedIntInterval.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal.clusternetworkaddressconfigprovider;
+
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * A non-empty integer interval. The index for each value is the 0-based position of the
+ * value in the interval from smallest to largest.
+ * @param startInclusive the inclusive start integer
+ * @param endExclusive the exclusive end integer
+ */
+record IndexedIntInterval(int startInclusive, int endExclusive) implements IndexedIntSet {
+
+    private static final String START_INCLUSIVE = "[";
```

Comment:
It depends how far you want to go. Are you talking just about the parser and how we name the characters in the notation, or rework the properties of the Interval too?

The JDK javadoc also describes if the bounds are inclusive/exclusive [repeatedly](https://docs.oracle.com/en%2Fjava%2Fjavase%2F21%2Fdocs%2Fapi%2F%2F/java.base/java/util/stream/IntStream.html#rangeClosed(int,int)). So I feel like the closed/open language is useful when describing types of Intervals but that inclusive/exclusive as a characteristic of the endpoint is easier for developers to understand without having to find a definition.

However, maybe when parsing the interval notation it would be reasonable to talk about whether it's open or closed notation, like `[ -> LEFT_OPEN` etc. I feel uncomfortable with pushing that all the way to the properties of the Interval though.